### ISSUE TITLE
SentryProfilerState doesn't respect `debug` option setting

### ISSUE THREAD

**Issue by jboulter11** [getsentry/sentry-cocoa#3943]
### Platform

iOS

### Environment

Develop

### Installed

Manually

### Version

Commit 74cf23b2

### Did it work on previous versions?

Not sure

### Steps to Reproduce

1. stand up sentry with debug = false but tracing and profiling at 100% so you will profile your session
2. Restart the app until you begin profiling

### Expected Result

1. no debug logs are printed to the console

### Actual Result

SentryProfilerState obliterates the log output, uses printf directly instead of SentryLog.

### Are you willing to submit a PR?

Probably not

**Comment by jboulter11** [comment]
![Screenshot by Dropbox Capture](https://github.com/getsentry/sentry-cocoa/assets/6633472/dd74e521-d560-432b-a901-41a47d293d3a)
snippet from SentryProfilerState.mm

**Comment by kahest** [comment]
Hey @jboulter11 thanks for the report - we have removed these in release `8.22.0` ([PR](https://github.com/getsentry/sentry-cocoa/pull/3738/files)) - are you able to update to that version?

**Comment by jboulter11** [comment]
Yes, it's in my queue to update to use the new xcframeworks.  Thanks for letting me know about the fix.
